### PR TITLE
Introduce a small_sym_mat_eigenvalues

### PR DESCRIPTION
**Pull request by alexfikl**
This introduces and uses a `_small_sym_mat_eigenvalues` to remove the posibility of having a negative number in the square root when solving for the eigenvalues.

### CODE REVIEW

File: pytential/symbolic/primitives.py [hunk 1]
```diff
@@ -858,7 +858,7 @@ def _small_mat_eigenvalues(mat):
         det_mat = a * d - b * c
 
         # solutions to lambda**2 - tr(A) * lambda + det(A)
-        sqrt_discriminant = cse(sqrt(tr_mat**2 - 4*det_mat))
+        sqrt_discriminant = cse(sqrt(abs(tr_mat**2 - 4*det_mat)))
```
Comment (inducer) [inline pytential/symbolic/primitives.py:861]: - Is it better to `abs` or to saturate at 0? (My gut feeling is saturate.)

Comment (inducer) [reply pytential/symbolic/primitives.py:861]: Or maybe specialize to a symmetric eigensolve?

Comment (alexfikl) [reply pytential/symbolic/primitives.py]: Went with specializing for a symmetric matrix, in which case that whole thing can be written as a sum of squares and is always nicely positive.

I also checked where it was failing, and there were just 3 nodes that had a value of `-4.4e-16`, so definitely some floating point issues.

Comment (inducer) [reply pytential/symbolic/primitives.py]: Thanks for checking! That makes me less worried about this whole affair. And the symmetric eigenvalue thing seems like a reasonably clean and reliable way out of this.

**Comment by inducer** [comment]
LGTM. Thanks!

**Pull request merged**